crates/polars-plan/src/dsl/expr.rs

```
@@ -369,6 +369,38 @@ impl Expr {
             .get(root)
             .to_field_and_validate(schema, ctxt, expr_arena)
     }
+
+    /// Extract a constant usize from an expression.
+    pub fn extract_usize(&self) -> PolarsResult<Option<usize>> {
```

Not sure why this is a `Result<Option<usize>>`, I don't think it returns `Ok(None)` anywhere

---

You're right, that was a carryover from its earlier usage as a helper function for `array_width`. I've removed the Option in [97a5a98](https://github.com/pola-rs/polars/pull/20941/commits/97a5a98d1e80e6e851b99c9a029f803017c3066e).

---

I'm wondering those `.unwrap()`s are ok or if we should return an error.

---

Mapped the error in [bfdaa7f](https://github.com/pola-rs/polars/pull/20941/commits/bfdaa7fb8ad67864948ccc020e91383a1c3f534d).